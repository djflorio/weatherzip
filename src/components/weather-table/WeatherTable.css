.weather-table {
  max-width: 700px;
  margin: auto;
  text-align: center;
}

.weather-table__header {
  color: rgb(0, 92, 104);
  text-align: center;
  font-size: 1.3em;
  font-weight: 600;
  margin: 0 0 10px 0;
}

.weather-table__new {
  font-size: 1em;
  border-style: none;
  padding: 5px 15px;
  border-radius: 5px;
  margin-bottom: 10px;
  background-color: rgb(0, 92, 104);
  color: #3CD1E6;
  cursor: pointer;
  transition: 0.3s;
}

.weather-table__new:hover {
  background-color: #fefefe;
  color: rgb(0, 92, 104);
}

.weather-table__list {
  list-style: none;
  width: 100%;
  padding: 0;
  border-collapse: collapse;
  text-align: center;
}