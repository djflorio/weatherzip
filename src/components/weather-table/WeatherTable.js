// Third party
import React from 'react';
import PropTypes from 'prop-types';

// Assets
import './WeatherTable.css';

// Components
import Row from '../row/Row';


const WeatherTable = (props) => (
  <div className="weather-table">
    <p className="weather-table__header">
      5-Day Forecast for {props.city}
    </p>
    <button
      className="weather-table__new"
      onClick={props.onNewSearch}>
      Search Again
    </button>
    <table className="weather-table__list">
      <tbody>
      {
        props.data.map(row => (
          <Row
            key={row.dt}
            icon={"http://openweathermap.org/img/w/" + row.weather[0].icon + ".png"}
            units={props.units}
            time={row.dt_txt}
            weather={row.weather[0].description}
            temp={row.main.temp}
          />
        ))
      }
      </tbody>
    </table>
  </div>
);

WeatherTable.propTypes = {
  city: PropTypes.string.isRequired,
  units: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
  onNewSearch: PropTypes.func.isRequired
}

export default WeatherTable;