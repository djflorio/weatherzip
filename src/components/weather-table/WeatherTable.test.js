import WeatherTable from './WeatherTable';

it("renders correctly", () => {
  const wrapper = shallow(
    <WeatherTable
      city="New York"
      units="imperial"
      data={[]}
      onNewSearch={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("calls onNewSearch when new search is clicked", () => {
  const spy = sinon.spy();
  const wrapper = shallow(
    <WeatherTable
      city="New York"
      units="imperial"
      data={[]}
      onNewSearch={spy}
    />
  );

  wrapper.find(".weather-table__new").simulate("click");

  expect(spy.calledOnce).toBe(true);
});

it("renders city name", () => {
  const wrapper = shallow(
    <WeatherTable
      city="New York"
      units="imperial"
      data={[]}
      onNewSearch={() => {}}
    />
  );

  const header = wrapper.find(".weather-table__header").text();

  expect(header).toContain("New York");

});