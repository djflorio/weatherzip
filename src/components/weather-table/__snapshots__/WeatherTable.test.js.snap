// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="weather-table"
>
  <p
    className="weather-table__header"
  >
    5-Day Forecast for 
    New York
  </p>
  <button
    className="weather-table__new"
    onClick={[Function]}
  >
    Search Again
  </button>
  <table
    className="weather-table__list"
  >
    <tbody />
  </table>
</div>
`;
